packages/xrpl/src/client/index.ts

```diff
@@ -666,8 +668,45 @@ class Client extends EventEmitter<EventTypes> {
       promises.push(checkAccountDeleteBlockers(this, tx))
     }
 
-    return Promise.all(promises).then(() => tx)
+    // further manipulation of tx_ uses non-SubmittableTransaction types, hence we need a typecast to any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/consistent-type-assertions -- `any` type is required to perform non-protocol modifications to the JSON object
+    const tx_ = tx as any
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A transaction JSON must contain TransactionType field
+    if (tx_.TransactionType === 'Payment') {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- A Payment transaction JSON must contain Amount field
+      if (tx_.Amount == null && tx_.DeliverMax != null) {
+        // If only DeliverMax is provided, use it to populate the Amount field
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access -- ensure that the aliased fields are identical
+        tx_.Amount = tx_.DeliverMax
+      }
+
+      // If Amount is not identical to DeliverMax, throw an error
+      if (
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- v2 rippled Payment-transaction JSON could contain these fields
+        tx_.DeliverMax != null &&
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- v2 rippled Payment-transaction JSON could contain these fields
+        tx_.Amount != null &&
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- v2 rippled Payment-transaction JSON could contain these fields
+        tx_.Amount !== tx_.DeliverMax
+      ) {
+        throw new ValidationError(
+          'PaymentTransaction: Amount and DeliverMax fields must be identical when both are provided',
+        )
+      }
+
+      // remove the DeliverMax field
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- v2 rippled Payment-transaction could contain DeliverMax field
+      if (tx_.DeliverMax != null) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- v2 rippled Payment-transaction could contain DeliverMax field
+        delete tx_.DeliverMax
+      }
```

Comment:
`tx_.DeliverMax != null` is a common check with all these conditionals. Try to simplify it even further.

Reply:
fixed in [5f218b2](https://github.com/XRPLF/xrpl.js/pull/2689/commits/5f218b2086887c0ae4746391eebfdca2e30b5909)